### ISSUE TITLE
Accessibility audit

### ISSUE THREAD

**Issue by andybraren** [thinkerbit/acorn#43]
A huge collection of links with accessibility considerations that Acorn should be checked against.

## Relevant links

* https://gomakethings.com/building-accessible-websites-and-apps-is-a-moral-obligation/
* Consultants
  * [Leonir Watson](https://youtu.be/OUDV1gqs9GA)
  * https://tink.uk/
  * https://tetralogical.com/
* Assessment tools
  * [WebAIM WAVE](http://wave.webaim.org/)
  * [Accessibility Testing with pa11y](https://bitsofco.de/pa11y/)
  * https://accessibilityinsights.io/
* Communities
  * https://www.applevis.com/about
* Techniques
  * List of [iOS VoiceOver gestures](https://www.applevis.com/guides/ios-voiceover/complete-list-ios-gestures-available-voiceover-users)
  * [Web navigation with VoiceOver](https://www.apple.com/voiceover/info/guide/_1134.html)
* Considerations
  * Prepending numbers to multi-author discussion threads like the [WebAIM forum](https://www.applevis.com/forum/other-apple-chat/has-applevis-website-changed)
  * https://davatron5000.github.io/a11y-nutrition-cards/
  * https://www.24a11y.com/2018/i-threw-away-my-mouse/
* Screen Reader text class
* Screen Readers
  * ["Skip Navigation"](https://webaim.org/techniques/skipnav/) Links
  * Allows screen reader users to quickly skip to the main content.
  * Working example and implementation [here](https://gomakethings.com/hidden-content-for-better-a11y/), works well just hitting tab.
  * Read [hidden text](http://rianrietveld.com/2015/04/04/the-screen-reader-text-class-why-and-how/) to screen reader users with hidden spans. [WordPress has had this](https://make.wordpress.org/accessibility/2015/02/09/hiding-text-for-screen-readers-with-wordpress-core/) for a while. In [some cases](https://accessible360.com/accessible360-blog/use-aria-label-screen-reader-text/) an aria-label may be better.
  * [Hidden "Accessibility Feedback" link](https://gomakethings.com/hidden-content-for-better-a11y/#ask-for-feedback) to solicit suggestions for future improvements.

Implement [skip-link underlines](https://medium.com/@iamhiwelo/improving-text-readability-for-dyslexic-users-with-skip-ink-underlines-bf52a2f3426b) for dyslexic and all other users.
* https://gomakethings.com/improving-link-legibility-with-two-lines-of-css/
* https://gomakethings.com/another-way-to-make-links-more-accessible/

Use underlines on any links by default
* [Reasons why](http://adrianroselli.com/2016/06/on-link-underlines.html)

- https://24ways.org/2017/wcag-for-people-who-havent-read-them/
- https://github.com/jackdomleo7/Checka11y.css
- [10 guidelines to improve web accessibility](https://aerolab.co/blog/web-accessibility/) (color, zoom, alt, subtitles, semantics, markup, roles, hiding)
- https://www.filamentgroup.com/lab/a11y-links.html
- https://github.com/Heydon/inclusive-design-checklist
- https://webaccessibility.guide/
- https://blogs.adobe.com/creativecloud/making-your-website-design-accessible/
- http://www.karlgroves.com/2013/09/07/diagnostic-css-super-quick-web-accessibility-testing/
- http://accessibility.voxmedia.com/
- Skip navigation link for screen readers? Mentioned [here](https://www.taniarascia.com/a-practical-guide-to-search-engine-optimization-seo-with-google/).
- [Accessible autocomplete](https://designnotes.blog.gov.uk/2017/04/20/were-building-an-autocomplete/)
- [Web accessibility issues according to actual people with disabilities](https://axesslab.com/accessibility-according-to-pwd/)
- [Twitter feedback on accessibility issues](https://hugogiraudel.com/2017/07/02/accessibility-feedback/)
- [Active and toggled states should be given the right aria-label and not just a class](https://css-tricks.com/user-facing-state/)
- https://medium.com/@matuzo/writing-css-with-accessibility-in-mind-8514a0007939
- https://bitsofco.de/tools-for-developing-accessible-websites/
- http://www.uxbooth.com/articles/accessibility-visual-design/
- https://medium.com/alistapart/writing-html-with-accessibility-in-mind-a62026493412
- Underline links option within reading options
- https://medium.com/@matuzo/writing-javascript-with-accessibility-in-mind-a1f6a5f467b9
- https://christianheilmann.com/2015/12/04/a-quick-reminder-on-how-and-why-to-use-labels-in-forms-to-make-them-more-accessible/
- [Every image should have alt text](https://bitsofco.de/alternative-text-and-images/)
- https://css-tricks.com/some-things-about-alt-text/
- Lists
	- https://mobile.twitter.com/scottohara/status/1084087699793997830
- Alt text on images
	- https://axesslab.com/alt-texts/
		- Make sure they end in period
		- Guide writer to not say this is an image of X
		- Also related https://twitter.com/vavroom/status/1067579069682769920
- https://gomakethings.com/accessibility-according-to-people-with-actual-disabilities/
- https://alistapart.com/article/accessibility-whack-a-mole
- https://webkit.org/blog/7551/responsive-design-for-motion/
- [Make jump links focus the heading](https://css-tricks.com/focus-styles-non-interactive-elements/)
- [Checklist for various components](https://medium.com/@addyosmani/accessible-ui-components-for-the-web-39e727101a67)
- https://inclusive-components.design/menus-menu-buttons/
- https://medium.com/bread-crumbs/free-web-accessibility-tools-round-up-b83a33797789
- [Menu structure](https://www.w3.org/WAI/tutorials/menus/structure/)
- [Make sure the spacebar jumps between headings, even with a sticky header](https://adactio.com/journal/10877)
- [Using the HTML lang attribute](https://www.paciellogroup.com/blog/2016/06/using-the-html-lang-attribute/)
- Modals or dialog
    - http://www.scottohara.me/article/modal-window-revisited.html
    - https://medium.com/flexport-design/designing-confirmation-278d159723e
    - https://micromodal.now.sh/, accessible and pure js
    - https://github.com/edenspiekermann/a11y-dialog
    - https://keithjgrant.com/posts/2018/meet-the-new-dialog-element/
    - https://css-tricks.com/a-css-approach-to-trap-focus-inside-of-an-element/
    - http://edenspiekermann.github.io/a11y-dialog/example/
    - https://scotch.io/bar-talk/build-native-web-modals-using-the-dialog-element222
    - https://github.com/edenspiekermann/a11y-dialog
- Modal improvements
    + Accessibility: http://humaan.com/modaal/
    + [A better way of removing the listener on the close button](https://gomakethings.com/when-to-use-return-with-javascript/)
    + http://dev.edenspiekermann.com/2016/02/11/introducing-accessible-modal-dialog/
    + https://heydesigner.com/blog/best-practices-modals-overlays-dialog-windows/
    + http://valhead.com/2015/11/23/ui-animation-reviews-modals/
    + https://www.w3schools.com/bootstrap/bootstrap_ref_js_modal.asp
    + https://frend.co/components/dialogmodal/
    + [Brad Frost asking Twitter](https://twitter.com/brad_frost/status/804432283335655424) how to best detect "click outside"
        * [Wes Bos gave a solution](https://codepen.io/wesbos/pen/eByeqp)
- A [Google-made accessibility course](https://www.udacity.com/course/web-accessibility--ud891)
- https://www.marcozehe.de/2015/12/14/the-web-accessibility-basics/
- [Dementia considerations](https://www.smashingmagazine.com/2016/05/designing-a-dementia-friendly-website/)
- [Accessible SVGs](https://css-tricks.com/accessible-svgs/)
    - http://simplyaccessible.com/article/7-solutions-svgs/
- https://meyerweb.com/eric/thoughts/2018/01/08/how-do-i-increase-accessibility/
- https://meyerweb.com/eric/thoughts/2018/01/10/increasing-accessibility/
- https://meyerweb.com/eric/thoughts/2018/01/18/headings-and-labels/
- https://css-tricks.com/small-tweaks-can-make-huge-impact-websites-accessibility/
- https://axesslab.com/text-splitting/
- https://www.stefanjudis.com/useful-accessibility-resources/
- https://alistapart.com/article/diy-web-accessibility-blueprint
- https://css-tricks.com/focusing-on-focus-styles/
- https://www.smashingmagazine.com/2018/04/designing-accessibility-inclusion/

Font size defaults
- https://blog.marvelapp.com/body-text-small/
